js/read_dir_test.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { test, testPerm, assert, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+testPerm({ write: true }, function readDirSyncSuccess() {
+  const files = deno.readDirSync("tests/");
+
+  for (const file of files) {
+    assert(file.isFile() || file.isDirectory());
+
+    if (file.isDirectory()) {
+      assertEqual(file.name, "subdir");
+    }
+
+    if (file.name === "002_hello.ts") {
+      assertEqual(file.path, `tests/${file.name}`);
+    }
+  }
```

Comment:
Please add counters to the `subdir` and `002_hello.ts` conditions above so that we can check here 
```js
assert(counter === 2)
```